lib/Backends/CMakeLists.txt

```diff
@@ -16,6 +16,23 @@ foreach(object ${subdirs})
   endif()
 endforeach()
 
+# External backends
```

Comment:
we need to add some docs around having external backends to add themselves to
linked_factories and linked_device_managers.

Reply:
I've added some docs about external backend. The purpose is not to detail all aspects that are common with standard backend, but more to focus on external backend specific aspects. Nevertheless, I've also summarized some aspects in common with standard backends.